TWLight/resources/views.py

```diff
@@ -216,6 +216,19 @@ def get_queryset(self):
 
         return partner_list
 
+    def get(self, request, *args, **kwargs):
+        try:
+            partner = self.get_object()
+        except Http404:
+            # If partner object does not exists check if the partner's status is NOT_AVAILABLE.
+            partner_pk = self.kwargs.get("pk")
+            if Partner.even_not_available.filter(pk=partner_pk).exists():
+                # If it is NOT_AVAILABLE then show a custom 404 message to users.
+                raise Http404("This partner is currently not open for applications")
```

Comment:
This string should be marked for translation, with a translator comment.

Reply:
done thanks :)